### PR TITLE
Add feature to match users for PeerPrep

### PR DESCRIPTION
**Pull request by gohyongjing**
Add backend and front end code for front end

More set up is required to connect both, will update in next PR as this PR is getting big

### CODE REVIEW

File: microservices/matching-service/src/client.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  );
+
+  return consumerPromise;
+}
+
+function generateUuid() {
+  return Math.random().toString() +
+         Math.random().toString() +
+         Math.random().toString();
+}
```
Comment (rexcyrio) [inline microservices/matching-service/src/client.ts:52]: There are packages for uuid generation -- see [`nanoid`](https://www.npmjs.com/package/nanoid) or [`uuid`](https://www.npmjs.com/package/uuid)

File: microservices/front-end/src/components/matchingService/utility/client.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  );
+
+  return consumerPromise;
+}
+
+function generateUuid() {
+  return Math.random().toString() +
+         Math.random().toString() +
+         Math.random().toString();
+}
```
Comment (rexcyrio) [inline microservices/front-end/src/components/matchingService/utility/client.ts:51]: See my other comment on uuid generation

**Review by rexcyrio — CHANGES_REQUESTED** [review]

File: microservices/matching-service/src/client.ts (hunk 1 above)
Comment (gohyongjing) [reply microservices/matching-service/src/client.ts:52]: Resolved in [6d0c79c](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g37/pull/14/commits/6d0c79c8c7227c5be27d09e9a5485950b72de899)

File: microservices/front-end/src/components/matchingService/utility/client.ts (hunk 2 above)
Comment (gohyongjing) [reply microservices/front-end/src/components/matchingService/utility/client.ts]: Resolved in [6d0c79c](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g37/pull/14/commits/6d0c79c8c7227c5be27d09e9a5485950b72de899)

**Review by rexcyrio — APPROVED** [review]
LGTM :+1:

**Pull request merged**